### PR TITLE
Add obsoletion message when config has Performance/Rails cops in it

### PR DESCRIPTION
**Pull request by dvandersluis**
Ran into this today when helping someone update rubocop. Performance cops were still in their config but no longer existing, and we did not have an obsoletion for it.

This adds a new type of cop obsoletions, where cops are extracted to another gem. It checks that the gem is not installed before outputting the error message (using the same bundle parsing as #9168, which will rebase away when that is merged) to ensure that configs are not marked as obsolete if `rubocop-performance` is active.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
Probably you can add the `Rails` cops there as well. They were also extracted at some point from the main gem.

**Comment by dvandersluis** [comment]
@bbatsov updated to include Rails too now.

**Comment by dvandersluis** [comment]
Test failures appear to be unrelated.

**Comment by bbatsov** [comment]
> This config should probably be extracted to rubocop-rails and removed from here.

Yeah, definitely.

File: lib/rubocop/config_obsoletion.rb [hunk 1]
```diff
@@ -8,7 +8,8 @@ class ConfigObsoletion
     COP_RULE_CLASSES = {
       'renamed' => RenamedCop,
       'removed' => RemovedCop,
-      'split' => SplitCop
+      'split' => SplitCop,
```
Comment (bbatsov) [inline lib/rubocop/config_obsoletion.rb:11]: I just realized that in the spirit of puns we should change "removed" to "retired". 😆 Probably we can also do something about "transfers between departments". 😆

**Comment by bbatsov** [comment]
Great work!

File: lib/rubocop/config_obsoletion.rb (hunk 1 above)
Comment (dvandersluis) [reply lib/rubocop/config_obsoletion.rb:11]: Hahaha yes!

**Pull request merged**